### ISSUE TITLE
Great Idea With Unexpected Result

### ISSUE THREAD

**Issue by shellscape** [tribou/jest-serializer-path#23]
Firstly, wonderful concept and glad that someone has taken the time to do this. Thank you for that. 

Surprisingly though, using this has unexpected results as it modifies the object being tested after the `expect` has been executed. That's not at all optimal as objects can still be used down the pipe. If this project is going to be opinionated about that (not a bad thing necessarily) then baking some manual mechanism for restoring the object would be preferred.

This code illustrates the problem:

```js
it('should test', () => {

  const result = getSomeObj({ ... }); // { path: '/Users/user/code/myproject/lib' }

  expect(result).toMatchSnapshot();

  // at this point the snapshot written is correct, and result.path is '<PROJECT_ROOT>/myproject/lib'
  // however, the result object has now been mutated, and silently no less

  // the following will now fail as the path has been mutated and does not exist
  fs.writeFileSync(path.join(result.path, 'myfile'), 'myfile data', 'utf-8');

});
```

I'd wager that this is a common problem. We're exploring using this module as a default for the [webpack-contrib](https://github.com/webpack-contrib) org, which means it'd be part of every official webpack loader and supporting development module. Unfortunately in it's current form, mutating the tested object is going block our use since we do quite a bit after a snapshot test for many modules.

**Comment by chrisblossom** [comment]
I'd expect the original object not to be mutated as well. I'll write some tests to verify this and put a PR in to fix this behavior.

**Comment by shellscape** [comment]
@chrisblossom thank you for the speedy reply!

**Comment by shellscape** [comment]
@tribou humbly requesting a publish for this fix

**Comment by tribou** [comment]
No problem, @shellscape. Published in `v0.1.14`.

Also, thanks for bringing this issue to our attention, and thanks to @chrisblossom for the quick fix! Testing my own `webpack` configurations was the inspiration for this module, so I hope it can be helpful to the community.
